src/randoop/operation/ArrayCreation.java

```diff
@@ -120,45 +143,48 @@ public String toStringVerbose() {
   }
 
   /**
-   * Returns constraint to represent new reference to this statement,
-   * namely the receiver that is generated.
+   * {@inheritDoc}
+   * @return type of created array.
    */
+  @Override
   public Class<?> getOutputType() {
-    if (outputTypeCached == null) {
-      outputTypeCached = Array.newInstance(elementType, 0).getClass();
-    }
     return outputTypeCached;
   }
 
   /**
-   * Appends string representation of ArrayDeclarationInfo into b.
+   * {@inheritDoc}
    */
-  public void appendCode(Variable newVar, List<Variable> inputVars, StringBuilder b) {
+  @Override
+  public void appendCode(List<Variable> inputVars, StringBuilder b) {
     if (inputVars.size() > length)
       throw new IllegalArgumentException("Too many arguments:"
           + inputVars.size() + " capacity:" + length);
+    
+    //TODO fix naming in statement
     String declaringClass = this.elementType.getCanonicalName();
-    String var = Variable.classToVariableName(this.elementType) + "_array" + newVar.index;
+   // String var = Variable.classToVariableName(this.elementType) + "_array" + newVar.index;
       
-    b.append(declaringClass + "[] "
-             + var
-             + " = new " + declaringClass + "[] { ");
+    b.append("new " + declaringClass + "[] { ");
     for (int i = 0; i < inputVars.size(); i++) {
       if (i > 0)
         b.append(", ");
       
-      // In the short output format, statements like "int x = 3" are not added to a sequence; instead,
-      // the value (e.g. "3") is inserted directly added as arguments to method calls.
-      Operation statementCreatingVar = inputVars.get(i).getDeclaringStatement(); 
-      if (!GenInputsAbstract.long_format
-          && ExecutableSequence.canUseShortFormat(statementCreatingVar)) {
-        b.append(PrimitiveTypes.toCodeString(((NonreceiverTerm) statementCreatingVar).getValue()));
+      // In the short output format, statements like "int x = 3" are not added 
+      // to a sequence; instead, the value (e.g. "3") is inserted directly added 
+      // as arguments to method calls.
+      Statement statementCreatingVar = inputVars.get(i).getDeclaringStatement(); 
+      if (!GenInputsAbstract.long_format && 
+          statementCreatingVar.isPrimitiveInitialization() &&
+          !statementCreatingVar.isNullInitialization()) {
+        String shortForm = statementCreatingVar.getShortForm();
+        if (shortForm != null) {
```

Comment:
This code looks wrong to me.  If shortForm is null, then the statement doesn't get output at all.  I think if shortForm is null, then the long form should be used instead.
(As usual, same comment elsewhere in the diffs.)